pangeo_forge_orchestrator/model_builders.py

```diff
@@ -90,12 +115,18 @@ def make_table_cls(self) -> SQLModel:
         https://sqlmodel.tiangolo.com/tutorial/fastapi/multiple-models/#the-hero-table-model,
         the table model is the same as the base model, with the addition of the ``table=True`` class
         creation keyword and an ``id`` attribute of type ``Optional[int]`` set to a default value of
-        ``Field(default=None, primary_key=True)``.
+        ``Field(default=None, primary_key=True)``. If ``self.relations`` are provided, they are
+        defined on the table model as described in the SQLModel documentation:
+        https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/define-relationships-attributes/
         """
         cls_name = self.make_cls_name(self.base, "")
         attrs = dict(id=Field(default=None, primary_key=True))
         annotations = dict(id=Union[int, None])
         attrs.update(dict(__annotations__=annotations))
+        if self.relations:
+            for r in self.relations:
+                attrs.update({r.field: Relationship(back_populates=r.back_populates.split(".")[1])})
+                attrs.get("__annotations__").update({r.field: r.annotation})  # type: ignore
```

Comment:
Can you say in words what this does? How does it transform the `relations` field? Give a before and after example.

Reply:
[Plain language summary]

`.relations` is an optional list of `RelationBuilder`s. If present, it is used in the `.make_table_cls` method to add `sqlmodel.Relationship` attributes to the generated table model, as described in the SQLModel documentation for [declaring relationship attributes](https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/define-relationships-attributes/#declare-relationship-attributes) on a table model. Specifically, each relationship attribute requires a field name (`RelationBuilder.field`), a type annotation (`RelationBuilder.annotation`), and the name of the field in the related model to back-populate (`RelationBuilder.back_populates`). Note that the annotation is either the related table class object (in the case of a one-to-one relationship), or a `List[str]` where `str` is the `__name__` of the related table class object. Either annotation type can be `Optional`.

[Before]

A `RelationBuilder` enters this control flow with the following fields:

- `field`: a name for the field to set on the table model being generated in this method
- `annotation`: a type annotation for the field to be set on the table model being generated in this method
- `back_populates`: the name of the field _in the related model_

[After]

For each of the `RelationBuilder` objects in the list of `self.relations`, the table model model generated in this method has the following fields set on it by the `self.relations` control flow in this method:

```python
from sqlmodel import Relationship, SQLModel

class GeneratedTableModel(SQLModel, table=True):   # the model we're generating in this method

    # ... other fields exist on this model (generated outside this particular `self.relations` control flow)

    RelationBuilder.field: RelationBuilder.annotation = Relationship(back_populates=RelationBuilder.back_populates)

```

> A small caveat: Currently, the `sqlmodel.Relationship.back_populates` kwarg value is not a 1-to-1 with `RelationBuilder.back_populates`, but it _will be_ after the change discussed in https://github.com/pangeo-forge/pangeo-forge-orchestrator/pull/31#discussion_r789907629 is implemented.